### PR TITLE
New Year's Day of Doctests

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/multidimensional.jl [hunk 1]
```diff
@@ -1150,6 +1150,33 @@ hash(x::Prehashed) = x.hash
...
+[:, :, 2] =
+  true   true
+ false  false
+
+julia> unique(A)
+2-element Array{Bool,1}:
+  true
+ false
+
+julia> unique(A,1)
```
Comment (tkelman) [inline base/multidimensional.jl]: I don't see entirely what this does from this example

File: base/reduce.jl [hunk 2]
```diff
@@ -270,6 +270,19 @@ should be evaluated as `(1-2)-3` or `1-(2-3)`. Use [`foldl`](@ref) or
...
 be executed in groups. Future versions of Julia might change the algorithm. Note that the
 elements are not reordered if you use an ordered collection.
+
+# Examples
+
+```jldoctest
+julia> reduce(+, 1, [2; 3; 4])
+10
+
+julia> reduce(-, 1, [2; 3; 4])
```
Comment (tkelman) [inline base/reduce.jl]: the docstring says not to use non-associative operations?

File: base/reduce.jl [hunk 3]
```diff
@@ -270,6 +270,19 @@ should be evaluated as `(1-2)-3` or `1-(2-3)`. Use [`foldl`](@ref) or
 Some operations accumulate error, and parallelism will also be easier if the reduction can
 be executed in groups. Future versions of Julia might change the algorithm. Note that the
 elements are not reordered if you use an ordered collection.
+
+# Examples
+
+```jldoctest
+julia> reduce(+, 1, [2; 3; 4])
```
Comment (tkelman) [inline base/reduce.jl]: the "unspecified"-ness about `v0` means it's probably best practice to use 0 for addition

File: base/set.jl [hunk 4]
```diff
@@ -147,9 +162,20 @@ function unique(f::Callable, C)
...
+```jldoctest
+julia> a = [1; 2; 3]
+3-element Array{Int64,1}:
+ 1
+ 2
+ 3
 
-Return `true` if all values from `itr` are distinct when compared with `isequal`.
+julia> allunique([a, a])
+ false
```
Comment (tkelman) [inline base/set.jl]: is there a leading space in the output?

File: base/reduce.jl (hunk 2 above)
Comment (kshyatt) [inline base/reduce.jl]: Aren't `+` and `-` associative on `Int`?

File: base/set.jl [hunk 5]
```diff
@@ -147,9 +162,20 @@ function unique(f::Callable, C)
 end
 
 """
-    allunique(itr)
+    allunique(itr) -> Bool
+
+Return `true` if all values from `itr` are distinct when compared with [`isequal`](@ref).
```
Comment (tkelman) [inline base/set.jl]: Reminding me that I liked the `alldistinct` naming better, if we're even going to describe it that way in the docstring

File: base/reduce.jl (hunk 2 above)
Comment (tkelman) [inline base/reduce.jl]: `(2 - 3) - 4` vs `2 - (3 - 4)`

Comment (kshyatt) [inline base/reduce.jl]: Derp right sorry

**Comment by kshyatt** [comment]
Not sure what's up with the failing `reflection` test. Doesn't seem related. This good to go?

**Comment by kshyatt** [comment]
Bugging everyone again, can I merge this or should I look harder at that test?

**Comment by KristofferC** [comment]
The same error on the same worker exists on all the current Travis builds.

**Pull request merged**